paasta_tools/paasta_maintenance.py

```diff
@@ -472,6 +474,29 @@ def is_safe_to_kill(hostname):
     return is_host_drained(hostname) or is_host_past_maintenance_start(hostname)
 
 
+def is_hostname_local(hostname):
+    return hostname == 'localhost' or hostname == getfqdn() or hostname == gethostname()
+
+
+def is_safe_to_drain(hostname):
+    """Checks if a host has healthy tasks running locally that have low
+    replication in other places
+    :param hostname: hostname to check
+    :returns: True or False
+    """
+    if not is_hostname_local(hostname):
+        print (
+            "Due to the way is_safe_to_drain is implemented, it can only work "
```

Comment:
I was hoping to reuse `get_marathon_services_running_here_for_nerve()`, as they represent the ones that are going to be in smartstack in the first place, and cross-reference that with `check_synapse_replication`.

I don't want this call to be expensive, remember we are going to be calling it once per hour for every paasta slave.